vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_unblinded.h

```diff
@@ -31,49 +32,88 @@ class Unblinded {
   explicit Unblinded(bat_ledger::LedgerImpl* ledger);
   ~Unblinded();
 
-  void Start(const std::string& viewing_id);
+  void Initialize();
+
+  void Start(const std::string& contribution_id);
+
+  void OnTimer(uint32_t timer_id);
 
  private:
+  void OnGetNotCompletedContributions(
+      ledger::ContributionInfoList list);
+
+  void GetContributionInfoAndUnblindedTokens(
+      const std::string& contribution_id,
+      GetContributionInfoAndUnblindedTokensCallback callback);
+
   void OnUnblindedTokens(
-      const std::string& viewing_id,
-      ledger::UnblindedTokenList list);
+      ledger::UnblindedTokenList list,
+      const std::string& contribution_id,
+      GetContributionInfoAndUnblindedTokensCallback callback);
+
+  void OnGetContributionInfo(
+      ledger::ContributionInfoPtr contribution,
+      std::vector<ledger::UnblindedToken> list,
+      GetContributionInfoAndUnblindedTokensCallback callback);
+
+  void PrepareTokens(
+      ledger::ContributionInfoPtr contribution,
+      std::vector<ledger::UnblindedToken> list);
+
+  void PreparePublishers(
+      std::vector<ledger::UnblindedToken> list,
+      ledger::ContributionInfoPtr contribution);
+
+  ledger::ContributionPublisherList PrepareAutoContribution(
+      std::vector<ledger::UnblindedToken> list,
+      ledger::ContributionInfoPtr contribution);
 
-  void MakeContribution(
-      const std::string& viewing_id,
-      ledger::UnblindedTokenList list);
+  void OnPrepareAutoContribution(
+      const ledger::Result result,
+      const std::string& contribution_id);
+
+  void ProcessTokens(const std::string& contribution_id);
 
-  bool GetStatisticalVotingWinner(
-      double dart,
-      const ledger::ReconcileDirections& directions,
-      Winners* winners) const;
+  void OnProcessTokens(
+      ledger::ContributionInfoPtr contribution,
+      std::vector<ledger::UnblindedToken> list);
 
-  void PrepareAutoContribution(
-      const std::string& viewing_id,
-      ledger::UnblindedTokenList list);
+  void TokenProcessed(
+      const ledger::Result result,
+      const std::string& contribution_id,
+      const std::string& publisher_key);
+
+  void OnTokenProcessed(
+      const ledger::Result result,
+      const std::string& contribution_id);
 
-  void GetStatisticalVotingWinners(
-      uint32_t total_votes,
-      const ledger::ReconcileDirections& directions,
-      Winners* winners) const;
+  void CheckIfDone(ledger::ContributionInfoPtr contribution);
 
   void SendTokens(
       const std::string& publisher_key,
       const ledger::RewardsType type,
-      ledger::UnblindedTokenList list,
-      SendTokensCallback callback);
+      std::vector<ledger::UnblindedToken> list,
```

Comment:
Every function that contains std::vector<ledger::UnblindedToken> should be checked could we use const & to avoid copies

Reply:
done